### PR TITLE
100% coverage with bug fixes.

### PR DESCRIPTION
**Pull request by ikarienator**

### CODE REVIEW

File: test/functions/arrow-expression.js [hunk 1]
```diff
@@ -117,13 +117,6 @@ suite("Parser", function () {
             null,
             new Shift.LiteralNumericExpression(0)))));
 
-    testParse("(a,a)=>0", expr, new Shift.ArrowExpression([
```
Comment (michaelficarra) [inline test/functions/arrow-expression.js]: We still need a test like this. Can we just change the parameter name here?

**Comment by michaelficarra** [comment]
LGTM otherwise.

File: test/functions/arrow-expression.js (hunk 1 above)
Comment (ikarienator) [inline test/functions/arrow-expression.js]: We have a `(a,b) => 0 + 1` before.

Comment (michaelficarra) [inline test/functions/arrow-expression.js]: Okay. I didn't see it.

**Pull request merged**